### PR TITLE
Content restructure

### PR DESCRIPTION
**Pull request by crutchcorn**
This PR aims to restructure the `content/` folder following the structure in #532:
```
content/
└── {author}/
    ├── index.md
    ├── profile.png
    ├── posts/{post}/
    │   └── index.md
    └── collections/{collection}/
        ├── index.md
        └── posts/{post}/
            └── index.md
```

- Most data collections are now stored in a `Map<string, *Info[]>`
  - This stores each translation of an item in the array, mapped to its id/slug
  - This structure mostly shouldn't be interacted with directly, but through `utils/api.ts` - which also resolves the locale for each item.
- Any "*Meta" properties referencing other data types are removed/replaced with their respective queries
  - Several components now require separate `posts=` or `authors=` properties to be provided
  - The `/api/search` endpoint now returns a separate `Record<string, UnicornInfo>` rather than providing each unicorn with the post object
- `ExtendedPostInfo` and `ExtendedCollectionInfo` types are removed
  - likewise, the `rehype-unicorn-populate-post` and `rehype-unicorn-get-suggested-posts` transforms are removed
  - This mostly isolates the remark/rehype process from the rest of the post data, so it can be used separately from `Astro.glob` in a future PR.
- The `collection-mapping.ts` previously used for the FFG site has been removed/replaced with a `pageLayout: "none"` frontmatter (which just removes it from the `/collections/[slug]` paths).
- `CONTRIBUTING.md` has been updated to reflect the new file structure

### CODE REVIEW

**Comment by crutchcorn** [comment]
# Testing

- [ ] About page
    - [x] Contents
    - [x] Translations
    - [ ] Unicorn list (buggy, see below)
- [ ] Blog post page
    - [x] Contents
    - [x] Multi-author
         - [x] Social image tabs
    - [x] Author page links
    - [x] Social image
    - [x] Images
    - [x] Translations
    - [ ] Related posts
        - [ ] Authors list links (buggy)
        - [x] Title
    - [x] Videos
    - [x] Components
        - [x] Tabs
        - [x] Summary
        - [x] iFrame
        - [x] Table of contents
        - [x] File tree
- [x] Collections page
    - [x] Table of contents
    - [x] Social images tags
    - [x] Epub generation
    - [x] Author PFP
- [ ] Unicorn page
    - [x] PFP
    - [x] About
    - [x] Achievements
    - [x] Posts
         - [x] List
         - [x] Author links
- [ ] Search
    - [x] Collections
    - [x] Articles
    - [x] Filter by author
    - [x] Filter by tag
    - [x] Sort
    - [x] Show filter
    - [x] Mobile view
    - [x] Pagination
- [ ] Post list
    - [x] Pagination
    - [x] Post list (13 pages)
    - [x] Post image cards
        - [x] Contains in-blog-post code
    - [x] Author list links
    - [x] Excerpt for posts without desc
    - [x] Desc listed
- [ ] FFG page
    - [ ] Corbin PFP/link (buggy)
    - [ ] Xeno PFP/link (buggy)
- [x] Compare RSS.xml to today (needs exact match)

**Comment by crutchcorn** [comment]
image

Our About page is randomizing our contributors again

**Comment by crutchcorn** [comment]
Another bug; 

All related posts have the same author as the current article:

image

image

(magic DOM is not me)

**Comment by crutchcorn** [comment]
Images on the FFG site never load:

image
image
image

image

**Comment by crutchcorn** [comment]
BUG:

The following pages are missing from post list:

- `/posts/astro-heathcheck-monitor-blog-for-issues`
- `/posts/building-original-projects`
- `/posts/figma-compose-line-height`
- `/posts/new-to-webdev-dont-use-react`

**Comment by crutchcorn** [comment]
BUG: UU members not showing up images in FFG homepage:

image

File: build-scripts/search-index.ts [hunk 1]
```diff
@@ -17,7 +17,9 @@ const createPostIndex = () => {
 			{
 				name: "authorName",
 				getFn: (post) => {
-					return post.authorsMeta.map((author) => author.name).join(", ");
+					return post.authors
+						.map((id) => api.getUnicornById(id, post.locale))
+						.join(", ");
```
Comment (fennifith) [inline build-scripts/search-index.ts:22]:
```suggestion
					return post.authors
						.map((id) => api.getUnicornById(id, post.locale)!.name)
						.join(", ");
```
This looks like it's missing a `.name` prop

Comment (fennifith) [reply build-scripts/search-index.ts:22]: Looking at the built index, I think `authorHandles` needs inspecting as well - seems to generate a lot of empty/duplicate items i.e. `"crutchcorn, crutchcorn, crutchcorn, , , , , "`.

Comment (crutchcorn) [reply build-scripts/search-index.ts]: Fixed

**Review by fennifith — APPROVED** [review]

**Pull request merged**